pkg/internalapis/cnsvolumeoperationrequest/cnsvolumeoperationrequest.go

```diff
@@ -119,23 +105,120 @@ func InitVolumeOperationRequestInterface(ctx context.Context) (VolumeOperationRe
-	log.Error(err)
-	return err
+	if operationToStore == nil {
+		msg := "cannot store empty operation"
+		log.Error(msg)
+		return errors.New(msg)
+	}
+	log.Debugf("Storing CnsVolumeOperationRequest instance with spec %v", spew.Sdump(operationToStore))
+
+	operationDetailsToStore := convertToCnsVolumeOperationRequestDetails(*operationToStore.OperationDetails)
+	instance := &cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequest{}
+	instanceKey := client.ObjectKey{Name: operationToStore.Name, Namespace: csiconfig.DefaultCSINamespace}
+
+	if err := or.k8sclient.Get(ctx, instanceKey, instance); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Create new instance on API server if it doesnt exist. Implies that this is the first time this object is being stored.
+			newInstance := &cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      instanceKey.Name,
+					Namespace: instanceKey.Namespace,
+				},
+				Spec: cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequestSpec{
+					Name: instanceKey.Name,
+				},
+				Status: cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequestStatus{
+					VolumeID:              operationToStore.VolumeID,
+					SnapshotID:            operationToStore.SnapshotID,
+					Capacity:              operationToStore.Capacity,
+					FirstOperationDetails: *operationDetailsToStore,
+					LatestOperationDetails: []cnsvolumeoperationrequestv1alpha1.OperationDetails{
+						*operationDetailsToStore,
+					},
+				},
+			}
+			err = or.k8sclient.Create(ctx, newInstance)
+			if err != nil {
+				log.Errorf("failed to create CnsVolumeOperationRequest instance %s/%s with error: %v", instanceKey.Namespace, instanceKey.Name, err)
+				return err
+			}
+			log.Infof("Created CnsVolumeOperationRequest instance %s/%s with latest information for task with ID: %s", instanceKey.Namespace, instanceKey.Name, operationDetailsToStore.TaskID)
+			return nil
+		}
+		log.Errorf("failed to get CnsVolumeOperationRequest instance %s/%s with error: %v", instanceKey.Namespace, instanceKey.Name, err)
+		return err
+	}
+
+	// Create a deep copy since we modify the object.
+	updatedInstance := instance.DeepCopy()
+
+	// Modify VolumeID, SnapshotID and Capacity
+	updatedInstance.Status.VolumeID = operationToStore.VolumeID
+	updatedInstance.Status.SnapshotID = operationToStore.SnapshotID
+	updatedInstance.Status.Capacity = operationToStore.Capacity
+
+	// Modify FirstOperationDetails only if it doesnt exist or TaskID's match.
+	firstOp := instance.Status.FirstOperationDetails
+	if firstOp.TaskID == "" || firstOp.TaskID == operationToStore.OperationDetails.TaskID {
+		updatedInstance.Status.FirstOperationDetails = *operationDetailsToStore
+	}
+
+	operationExistsInList := false
+	// If the task details already exist in the Status, update it with the latest information.
+	for index, operationDetails := range instance.Status.LatestOperationDetails {
```

Comment:
Can we iterate `LatestOperationDetails` in reverse order? This way you will find the task sooner.